AK/URL.h

```diff
@@ -47,6 +48,10 @@ class URL {
         : URL(string.view())
     {
     }
+    URL(String const& string)
+        : URL(string.bytes_as_string_view())
+    {
+    }
```

Comment:
Why? `String` should be able to implicitly convert to `StringView` and invoke the appropriate constructor above

https://github.com/SerenityOS/serenity/blob/8f1d13e73b8bae4f86e5cf2ed7c42d757a6ac0a7/AK/StringView.h#L51

Reply:
Without it the compiler complains about the conversions listed below.

https://github.com/kennethmyhra/serenity/blob/a658386b38735d7b834b4516456c48978f9e4879/Userland/Libraries/LibWeb/URL/URL.cpp#L30

https://github.com/kennethmyhra/serenity/blob/a658386b38735d7b834b4516456c48978f9e4879/Userland/Libraries/LibWeb/URL/URL.cpp#L40

https://github.com/kennethmyhra/serenity/blob/a658386b38735d7b834b4516456c48978f9e4879/Userland/Libraries/LibWeb/URL/URL.cpp#L101

I can get around it by doing this:
```cpp
// 1. Let parsedBase be the result of running the basic URL parser on base.
parsed_base = AK::URL { base.value() };
```
or this:
```cpp
// 1. Let parsedBase be the result of running the basic URL parser on base.
parsed_base = base.value().bytes_as_string_view();
```

Reply:
I see - not a huge fan of implicit URL parsing anyway, but let's preserve the status quo for now 👍